Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -240,6 +222,44 @@ public struct BSONDocument {
         set { self[member] = newValue }
     }
 
+    /**
+     * Returns a copy of this document with an `_id` element prepended. If the document already contains an `_id`,
+     * returns the document as-is.
+     * - Throws: `BSONError.DocumentTooLargeError` if adding the `_id` would make the document exceed the maximum
+     *           allowed size for a document.
+     * - SeeAlso: https://docs.mongodb.com/manual/core/document/#the-id-field
+     */
+    public func withID() throws -> BSONDocument {
```

Comment:
Where are the other places that we throw for the document becoming too large? I wonder if it's rare enough that we ought to just fatalError like we do in the subscripts. Given that most users are using BSON with the driver / db, the limit in practice is what the server can take (16MB), so the ~2GB `MAX_BSON_SIZE` is never really hit.

Reply:
the only other place we throw this error is in `set(key:to:)`, which is the underlying implementation for the subscript setter method. so it ultimately always gets propagated as a `fatalError` via that path.
I guess it does seem inconsistent that this would throw but subscript doesn't, but OTOH we are actually able to propagate the error the user in the place we call this from in the driver so it is probably beneficial to do so.